File: Strategy/problem.dart
class ImageStorage {
  String compressor;
  String filter;

  ImageStorage({
    required this.compressor,
    required this.filter,
  });

  void store(String fileName) {
    if (compressor == "jpeg") {
      print("Compressing using JPEG");
    } else if (compressor == "png") {
      print("Compressing using PNG");
    }

    if (filter == "b&w") {
      print("Applying B&W filter");
    } else if (filter == "high-contrast") {
      print("Applying high contrast filter");
    }
  }
}

File: Strategy/problem.py
class ImageStorage:
    def __init__(self, compressor, filter) -> None:
        self.compressor = compressor
        self.filter = filter

    def store(self, fileName: str) -> None:
        if self.compressor == "jpeg":
            print(f"{fileName}: Compressing using JPEG")
        elif self.compressor == "png":
            print(f"{fileName}: Compressing using PNG")

        if self.filter == "b&w":
            print(f"{fileName}: Applying B&W filter")
        elif self.filter == "high-contrast":
            print(f"{fileName}: Applying high contrast filter")
